pkg/rancher-components/src/components/StringList/StringList.vue

```diff
@@ -0,0 +1,573 @@
+      selected:     null as string | null,
+      isEditItem:   null as string | null,
+      isCreateItem: false,
+      errors:       { duplicate: false } as Record<Error, boolean>,
+    };
+  },
+
+  computed: {
+
+    /**
+     * Create an array of error messages, one for each current error
+     */
+    errorMessages(): string[] {
+      return Object
+        .keys(this.errors)
+        .filter(f => !!(this.errors)[f as Error])
+        .map(k => this.messages[k]);
+    },
+  },
+
+  watch: {
+    /**
+     * Shutdown any user actions on the component when it becomes readonly
+     */
+    readonly() {
+      this.toggleEditMode(false);
+      this.toggleCreateMode(false);
+    },
+  },
+
+  methods: {
+    onChange(value: string) {
+      this.value = value;
+      /**
+       * Remove duplicate error when a new value is typed
+       */
+      this.toggleError(
+        'duplicate',
+        false,
+        this.isCreateItem ? INPUT.create : INPUT.edit,
+      );
+    },
+
+    onSelect(item: string) {
+      if (this.isCreateItem || this.isEditItem === item) {
+        return;
+      }
+      this.selected = item;
+    },
+
+    onSelectNext(arrow: Arrow) {
+      const index = findIndex(
+        this.items as string[],
+        this.selected as string,
+      );
+
+      if (index !== -1) {
+        /**
+         * Select next item in the arrow's direction if it exists,
+         * else select again this.selected (do nothing)
+         */
+        const item = (this.items[index + DIRECTION[arrow]] || this.selected) as string;
+
+        this.onSelect(item);
+        this.moveScrollbar(arrow);
+      }
+    },
+
+    onSelectLeave() {
+      this.selected = null;
+    },
+
+    onClickEmptyBody() {
+      if (!this.isCreateItem && !this.isEditItem) {
+        this.toggleCreateMode(true);
+      }
+    },
+
+    onClickPlusButton() {
+      this.onSelectLeave();
+      this.toggleCreateMode(true);
+    },
+
+    onClickMinusButton() {
+      if (this.isCreateItem) {
+        this.toggleCreateMode(false);
+
+        return;
+      }
+      if (this.isEditItem) {
+        this.toggleEditMode(false);
+
+        return;
+      }
+      if (this.selected) {
+        const index = this.deleteItem(this.selected);
+
+        if (index !== -1) {
+          /**
+           * Select the next item in the list when an item is deleted.
+           * If the deleted item was the last item, select the previous item (this.items[index - 1]). 
+           */
+          setTimeout(() => {
+            const item = (this.items[index] || this.items[index - 1]) as string;
+
+            this.onSelect(item);
+            this.setFocus(item);
+          }, 100);
+        }
+      }
+    },
+
+    setFocus(refId: string) {
+      setTimeout(() => {
+        this.getRef(refId)?.focus();
+      }, 0);
+    },
+
+    /**
+     * Move scrollbar when the selected item is over the top or bottom side of the box
+     */
+    moveScrollbar(arrow: Arrow, value = 25) {
```

Comment:
Why is value default 25? What is aimed to?

Reply:
25 is the height of each div in the list box, in px:

```
    .static {
      height: 25px;
      padding: 3px;
    }
```

Reply:
Should we try to get the ref height instead so we do not have to hard code it??